### DISCUSSION TITLE
How to stop creating spans under the same trace?

### DISCUSSION THREAD

**Discussion by GokulShrikanth** [open-telemetry/opentelemetry-java-instrumentation#6208] [Q&A]
Hi guys, 
I am continuously polling a message queue(RMQ) inside a vertx timer, This is causing the issue of making all the spans to be under the same trace ID whereas I want the individual messages to be different traces. Is there a way to break the context and create traces for every message individually?

**Comment by mateuszrzeszutek** [comment]
Hey @GokulShrikanth ,
Are you using plain OTel `Tracer`/[`SpanBuilder`](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/SpanBuilder.java) API? Or opentelemetry-instrumentation-api's `Instrumenter`?
If it's the former, you can use `setParent()`/`setNoParent()` to explicitly set the parent context - either one with a span extracted from incoming message, or one without a span whatsoever.
If you're using [`Instrumenter`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java) you can pass the context of your choosing to the `start()` method - again, one extracted from the message, or `Context.root()` (which is empty).

**Reply by GokulShrikanth** [reply]
Hi @mateuszrzeszutek ,
I am using the **Tracer/SpanBuilder** API's.. But I am not creating any manual spans over the method I use for polling, I am sharing the snipet of code where I face this issue below

```java
private void getNewRequest() {
    String res = RabbitmqBroker.getMessageFromQueue();
    if (res != null) {
        try {
            processMsg(res);
        } catch (Exception exception) {
           logException(exception, "getNewRequest");
        }
        getNewRequest();
    } else {
        getVertx().setTimer(6000, id -> {
            getNewRequest();
        });
    }
}
```

**Reply by mateuszrzeszutek** [reply]
Hm, I see - I suppose the `RabbitmqBroker.getMessageFromQueue()` call does the `basicGet()` call underneath.

You could forcefully suppress the context propagation by making `Context.root()` the current context for the execution of this method:

```java
private void getNewRequest() {
    try (Scope ignored = Context.root()) {
        String res = RabbitmqBroker.getMessageFromQueue();
        if (res != null) {
            try {
                processMsg(res);
            } catch (Exception exception) {
                logException(exception, "getNewRequest");
            }
            getNewRequest();
        } else {
            getVertx().setTimer(6000, id -> {
                getNewRequest();
            });
        }
    }
}
```

This way the RabbitMQ instrumentation will inherit the empty context (`Context.root()`) instead of the context of the caller of `getNewRequest()`.
